datafusion/sql/src/unparser/expr.rs

```diff
@@ -204,6 +245,12 @@ impl Unparser<'_> {
                         end_bound: Option::from(end_bound),
                     }),
                 }));
+
+                let order_by_expr_vec: Vec<ast::OrderByExpr> = order_by
+                    .iter()
+                    .flat_map(|expr| expr_to_unparsed(expr)?.into_order_by_expr())
+                    .collect();
```

Comment:
I think this should be just `map` (flat map I think will discard Result`s 🤔 

```suggestion
                let order_by: Vec<ast::OrderByExpr> = order_by
                    .iter()
                    .map(|expr| expr_to_unparsed(expr)?.into_order_by_expr())
                    .collect()?;
```